[Enterprise Search][Workplace Search] Migrate Groups to Kibana

scottybollinger opened the pull request:
## Summary

This PR converts the first major component tree, Groups, to Kibana.

The strategy was to take the existing code and paste it directly into Kibana, with the only changes being linting issues, such as formatting and semicolons, along with some lint nits like default exports fixed.

The reasoning behind this is that the existing code works in `ent-search` and this prevent the need to completely re-review the entire app and only need to review the changes needed to get the app to work in Kibana.

The rest of the PR is a commit by commit migration of the components and containers themselves.

![image](https://user-images.githubusercontent.com/1869731/94190090-3f996b00-fe71-11ea-9f62-a1eb5eeeda3a.png)

![image](https://user-images.githubusercontent.com/1869731/94190174-622b8400-fe71-11ea-888d-76517c9d587c.png)

![image](https://user-images.githubusercontent.com/1869731/94190566-e251e980-fe71-11ea-8759-e85c022abc10.png)

### Note about i18n and unit tests
Because of the size of this PR is ~3000 I am going to go back and add i18n and unit tests in a future PR. 

### Known issues

- Page title switches to App Search when child route is refreshed
- Header nav (search link on right) disappears when navigating back to Group overview page from child
- Queued Flash Messages don't work as expected (show up on next route change after desired)
- Breadcrumbs need to be fixed to include 4 levels

### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

scottybollinger committed:
Initial copying of Groups component tree

This commit moves the base component tree from ent-search with the following changes to satisfy pre-commit hooks

- All file names changed to camel_case
- Copyright comment added to the top of each file
- Semicolons and formatting to match Kibana
- Default exports removed from components
- Placeholder keyboard listener functions added to non-interactive elements with click handlers

scottybollinger committed:
Update types

- Adds new types
- Moves types needed by server and app to common

scottybollinger committed:
Add constants and image

scottybollinger committed:
Update paths, remove kea typecasting

This commit does the following:

- Updates the paths of the imports
- Removed the need to typecast logic i.e. "as IGroupsLogic" after the Kea hooks
- Fixed a few TypeScript errors with events and return values on functions

- Finally, this commit does away with the ConfirmModal component (was never copied over), as it only wraps 2 EUI components and we decided it was not needed

scottybollinger committed:
Refactor ContentSection and ViewContentHeader

With the groups components, we needed to add a header action that was right-aligned to ContentSection.

After building what was needed, I realized that the header in the ContentSection was basically a ViewContentHeader with a different sized heading. I refactored ViewContentHeader to have variable sizes and for ContentSection to use that instead.

Also added styles from ent-search

scottybollinger committed:
Add group routes

This is both the server routes and the frontend paths. After conversation with Jonas, we decided to not use the `/org` prefix. Will clean up the other routes as we build out the rest the migration

scottybollinger committed:
Update logic files

This commit converts the logic files for use in Kibana. This includes:

- Using the new Kea 2.2 syntax that uses MakeLogicType to provide types
- Uses the Kibana HttpLogic instead of the rails routes/http
- Adds return types to Actions interfaces
- Adds `path` key to help with debuggin in dev tools
- Removes FashMessages in lieu of separate logic. Because of this, we have to manually clear the flash mesages with listeners where reducers used to do it when the messages were local to the logic file
- Preplaces promises with async/await & try/catch

- Also, in GroupsLogic’s getSearchResults method, we used Kea’s breakpoint functionality and replaced the useDebouce Hook that was used in the component (future commit)
- Uses global lodash per Kibana’s new directive

scottybollinger committed:
Update routers an indexes

This PR configures the routers to work with Kibana

- Updates paths to imports
- Adds top-level styles

For GroupRouter
- Removes AppView
- Use global flash messages
- Remove sidebar and breadcrumbs

scottybollinger committed:
Update GroupOverview

Adds some changes to facilitate the new design for Kibana

- Copy changes
- Layout changes to have buttons inline and not conditionally shown

scottybollinger committed:
Various updates to components

Adds some changes to facilitate the new design for Kibana.

- Remove unnecessary TableHeader
- Adds pencil in lieu of manage button per design

scottybollinger committed:
Update main groups component

A previous commit did this for the components, as the intention was to do this for components in one PR and the others an a separate PR. Unfortunately the build does not pass with all the missing imports.

This commit does the following:
- Updates the paths of the imports
- Removed the need to typecast logic i.e. "as IGroupsLogic" after the Kea hooks
- Fixed a few TypeScript errors with events and return values on functions
- Use global flash messages
- Remove debounces filderValue, as it’s now debounces in logic file
- Remove legacy isCurated props
- Remove legact AppView

scottybollinger committed:
Add sub navigation to main nav

scottybollinger committed:
Remove extra search app link

The link is now in the header and is not needed in the sidebar

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/)
* Commit: 9d5fe105e650a52c49ad6ef916b27b2892622e54
* This comment will update when the build is complete

### Failed CI Steps
* [Lint: eslint](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/execution/node/348/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/","number":"77082","commit":"9d5fe105e650a52c49ad6ef916b27b2892622e54"}]}
PIPELINE-->

scottybollinger commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into scottybollinger/groups

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/)
* Commit: 8c2d5d4571d58906216931d2f6ba021b188f4d6e
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/execution/node/366/log)

### History
* :broken_heart: [Build #77082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/) failed 9d5fe105e650a52c49ad6ef916b27b2892622e54

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/","number":"77082","commit":"9d5fe105e650a52c49ad6ef916b27b2892622e54"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/","number":"77501","commit":"8c2d5d4571d58906216931d2f6ba021b188f4d6e"}]}
PIPELINE-->

scottybollinger commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into scottybollinger/groups

nreese committed:
Add deprecated message to tile_map and region_map visualizations. (#77683)

* Add deprecation message to coordinate map and region map

* clean up text

* add default distro link and view in maps link

* move url generation into onClick handler

* create tile map layer descritor

* set metrics and color and scaling

* lazy load createTileMapLayerDescriptor

* tslint fixes

* tslint cleanup for OSS code

* add region map deprecation message

* tslint cleanup

* consolidate logic into LegacyMapDeprecationMessage

* fix jest test

* fix tile-map and region_map in OSS distro

* tslint fixes

* assert urlGenerator exists

* update message text

* ensure legacy-ids get correctly evaluated (#37)

* handle 6.x region map saved objects

* turn off field meta

* fix type

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Thomas Neirynck <thomas@elastic.co>

smith committed:
Fix APM lodash imports (#78438)

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
[kbn/optimizer] fix .json extension handling (#78524)

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Commit:
[kbn/es] use a basic build process (#78090)

Co-authored-by: spalger <spalger@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

jonathan-buttner committed:
[Security Solution] Initiate endpoint package upgrade from security app (#77498)

* Working on package update functionality

* Correctly installing package

* Moving upgrade component and working upgrade

* Doing permissions check

* Cleaning up imports

* Adding bulk upgrade api

* Addressing comments

* Removing todo

* Changing body field

* Adding helper for getting the bulk install route

* Adding request spec

* Using bulk install endpoint from ingest

* Moving component to a hook

* Addressing feedback

bhavyarm committed:
Fixing a11y test failure on discover app (https://github.com/elastic/kibana/issues/59975) (#77614)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77569/)
* Commit: 3b0e58eed187d65c65006bce1891f807655737c8

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/3b0e58eed187d65c65006bce1891f807655737c8) | diff | [baseline](https://github.com/elastic/kibana/commit/c285287da39abe4c9f091c02f5ae3ffd10ca25a7) |
| --- | --- | --- | --- |
| enterpriseSearch | 399 | +152 | 247 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/3b0e58eed187d65c65006bce1891f807655737c8) | diff | [baseline](https://github.com/elastic/kibana/commit/c285287da39abe4c9f091c02f5ae3ffd10ca25a7) |
| --- | --- | --- | --- |
| enterpriseSearch | 705.1KB | :warning: +215.8KB | 489.2KB |

#### miscellaneous assets size
| id | [value](https://github.com/elastic/kibana/commit/3b0e58eed187d65c65006bce1891f807655737c8) | diff | [baseline](https://github.com/elastic/kibana/commit/c285287da39abe4c9f091c02f5ae3ffd10ca25a7) |
| --- | --- | --- | --- |
| enterpriseSearch | 843.2KB | +8.4KB | 834.9KB |

#### distributable file count
| id | [value](https://github.com/elastic/kibana/commit/3b0e58eed187d65c65006bce1891f807655737c8) | diff | [baseline](https://github.com/elastic/kibana/commit/c285287da39abe4c9f091c02f5ae3ffd10ca25a7) |
| --- | --- | --- | --- |
| default | 45673 | +2 | 45671 |

### History
* :broken_heart: [Build #77501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/) failed 8c2d5d4571d58906216931d2f6ba021b188f4d6e
* :broken_heart: [Build #77082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/) failed 9d5fe105e650a52c49ad6ef916b27b2892622e54

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/","number":"77082","commit":"9d5fe105e650a52c49ad6ef916b27b2892622e54"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/","number":"77501","commit":"8c2d5d4571d58906216931d2f6ba021b188f4d6e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77569/","number":"77569","commit":"3b0e58eed187d65c65006bce1891f807655737c8"}]}
PIPELINE-->

pgayvallet committed:
fix createAppNavigationHandler to use `navigateToUrl` (#78583)

alvarezmelissa87 committed:
[ML] DF Analytics creation: ensure job did not fail to start before showing results link (#78200)

* ensure job did not fail to start before showing results link

* simplify job started check

* update job finished check

Commit:
[Enterprise Search] Remove all instances of KibanaContext to Kea store (#78513)

* Add KibanaLogic store/mount

* Update usage of setBreadcrumbs/setDocTitle to KibanaLogic

* Update usage of Kibana config context to KibanaLogic

* Update usage of navigateToUrl context to KibanaLogic

* :fire: Remove unused shallow_usecontext mocks, repurpose file to shallow_useeffect

- The file is now no longer used to mock useContext, only unseEffect, hence the rename

* :fire: Delete/repurpose mount with context helpers

- Remove mountWithKibanaContext completely
- Change mountWithContext to mountWithIntl, since that's the only context remaining, and move it to its own file
- Change mountWithAsyncContext to just mountAsync and move it to its own file
  + add an option to pair it w/ mountWithIntl for formatted date/number support

* :fire: Delete KibanaContext and mockKibanaContext

+ minor newline linting/grouping tweaks

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

scottybollinger commented:
Sorry for anyone this pinged :(

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77635/)
* Commit: 26d6d79f543237498e54f7fb7895eb13a31a98f3
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77635/execution/node/350/log)

### History
* :green_heart: [Build #77569](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77569/) succeeded 3b0e58eed187d65c65006bce1891f807655737c8
* :broken_heart: [Build #77501](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/) failed 8c2d5d4571d58906216931d2f6ba021b188f4d6e
* :broken_heart: [Build #77082](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/) failed 9d5fe105e650a52c49ad6ef916b27b2892622e54

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77082/","number":"77082","commit":"9d5fe105e650a52c49ad6ef916b27b2892622e54"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77501/","number":"77501","commit":"8c2d5d4571d58906216931d2f6ba021b188f4d6e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77569/","number":"77569","commit":"3b0e58eed187d65c65006bce1891f807655737c8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/77635/","number":"77635","commit":"26d6d79f543237498e54f7fb7895eb13a31a98f3"}]}
PIPELINE-->
